File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MaxLength.java
package com.jiangtj.micro.common.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Documented
@Constraint(validatedBy = MaxLengthValidator.class)
@Target({METHOD, FIELD, PARAMETER})
@Retention(RUNTIME)
public @interface MaxLength {
    int value();

    boolean trim() default true;

    String message() default "最小字符串长度不能大于 {value}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MaxLength.kt
package com.jiangtj.micro.common.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [MaxLengthValidator::class])
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.VALUE_PARAMETER
)
@Retention(AnnotationRetention.RUNTIME)
annotation class MaxLength(
    val value: Int,
    val trim: Boolean = false,
    val message: String = "最小字符串长度不能大于 {value}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
